src/verifiers/issuerIdentity/builder.ts

```diff
@@ -64,6 +39,8 @@ const defaultVerifier: IssuerIdentityVerifier = async ({ issuerIndex }) => {
 export const issuerIdentityVerifierBuilder = (verifiers: IssuerIdentityVerifierDefinition[]): VerifierType => {
   const registeredVerifiers: { [identityProofType: string]: IssuerIdentityVerifier } = {};
   verifiers.forEach((verifier) => {
+    if (registeredVerifiers[verifier.type])
+      throw new Error(`Cannot register verifier to a type ${verifier.type} which has existing verifier assigned`);
     registeredVerifiers[verifier.type] = verifier.verify;
   });
 
```

Comment:
if .test() is always true then skipFragment should never be called if skipFragment is called it implies something has gone wrong so perhaps it should throw an error to indicate that the world has ended in an unexpected way

Reply:
Using the verification based on the type makes it completely invalid / unusable for OpenCerts:

- the registry verifier run for all documents
- DNS-TXT may run as well for this documents

That means one issuer may have 2 verifications but this implementation doesn't allow for it. Registry verifier has priority over DNS-TXT. A new institution will need to use DNS-TXT as long as they are not part of the registry so this will be needed forever.

On top of that there is the case where one verifier may fail.

Reply:
On OpenCerts, the method will be to:

1. Not use the DNS-TXT module. Instead create a module that inherits from the DNS-TXT and do an OR with the OpenCerts Registry:

```
  const dnsTxtResults = verify({document, issuerIndex, options});
  const registryResults = getRegistryIdentity(documentStore);
  return {
    status: valid if any of result is valid,
    identifier: registryResult takes precedence, else dnsTxtResult's identity
  }
```

2. Export this new identity provider

```
export const OpenCertsDnsTxtIdentityProof: IssuerIdentityVerifierDefinition = {
  type: "OC-DNS-TXT",
  verify,
};
```

3. Use it:

```
  const identityVerifier = issuerIdentityVerifierBuilder([OpenCertsDnsTxtIdentityProof, OpenAttestationDnsDidIdentityProof]);
  const verify = verificationBuilder([
  openAttestationHash,
  openAttestationEthereumTokenRegistryStatus,
  openAttestationEthereumDocumentStoreStatus,
  OpenAttestationDidSignedDocumentStatus,
  identityVerifier
])
```